### PR TITLE
r1cs-std works with no-std

### PR DESCRIPTION
**Pull request by debris**
changes:
- `test-constraint-system` uses `BTreeMap` instead of `radix_trie`
- `r1cs-std` supports `no-std`

### CODE REVIEW

**Comment by Pratyush** [comment]
Hmm for the radix-trie change you can just replace it with a standard map type, like BTreeMap. (I'd prefer to avoid splitting it out into a separate crate.)

File: r1cs-std/src/test_constraint_system.rs [hunk 1]
```diff
@@ -63,6 +62,7 @@ impl<ConstraintF: Field> TestConstraintSystem<ConstraintF> {
         }
     }
 
+    #[allow(unused_variables)]
```
Comment (Pratyush) [inline r1cs-std/src/test_constraint_system.rs:65]: Why is this needed here?

Comment (debris) [reply r1cs-std/src/test_constraint_system.rs:65]: a redefinition of `println` on `no-std` doesn't do anything

Comment (Pratyush) [reply r1cs-std/src/test_constraint_system.rs:65]: ahh so the idea is that since the no_std version of the macro doesn't do anything with the variable, it leads to the variable being left unused. Ok, makes sense.
